pkg/solana/capabilities/targets/write_target.go

```diff
@@ -0,0 +1,214 @@
+	// generate ID based on chain selector
+	name := fmt.Sprintf("write_solana_%v", chain.ID())
+
+	info := capabilities.MustNewCapabilityInfo(
+		name,
+		capabilities.CapabilityTypeTarget,
+		"Write target.",
+		"v1.0.0",
+	)
+
+	reader, err := chain.Reader()
+	if err != nil {
+		return nil, err
+	}
+
+	return &SolanaWrite{
+		chain,
+		reader,
+		info,
+		lggr,
+	}, nil
+}
+
+type SolanaAccount struct {
+	PublicKey  string `mapstructure:"public_key"`
+	IsWritable bool   `mapstructure:"is_writable"`
+	IsSigner   bool   `mapstructure:"is_signer"`
+}
+
+type SolanaConfig struct {
+	ChainID           uint
+	ReceiverProgramID sdk.PublicKey
+	Params            []any
+	Accounts          []SolanaAccount
+	// ABI               string TODO:
+}
+
+// TODO: enforce required key presence
+
+func parseConfig(rawConfig *values.Map) (SolanaConfig, error) {
+	var config SolanaConfig
+	configAny, err := rawConfig.Unwrap()
+	if err != nil {
+		return config, err
+	}
+	err = mapstructure.Decode(configAny, &config)
+	return config, err
+}
+
+// func evaluateParams(params []any, inputs map[string]any) ([]any, error) {
+// 	vars := pipeline.NewVarsFrom(inputs)
+// 	var args []any
+// 	for _, param := range params {
+// 		switch v := param.(type) {
+// 		case string:
+// 			val, err := pipeline.VarExpr(v, vars)()
+// 			if err == nil {
+// 				args = append(args, val)
+// 			} else if errors.Is(errors.Cause(err), pipeline.ErrParameterEmpty) {
+// 				args = append(args, param)
+// 			} else {
+// 				return args, err
+// 			}
+// 		default:
+// 			args = append(args, param)
+// 		}
+// 	}
+
+// 	return args, nil
+// }
+
+func (cap *SolanaWrite) Execute(ctx context.Context, callback chan<- capabilities.CapabilityResponse, request capabilities.CapabilityRequest) error {
+	cap.lggr.Debugw("Execute", "request", request)
+
+	config := cap.chain.Config().ChainWriter()
+
+	if config == nil {
+		return fmt.Errorf("ChainWriter config undefined")
+	}
+
+	reqConfig, err := parseConfig(request.Config)
+	if err != nil {
+		return err
+	}
+
+	inputsAny, err := request.Inputs.Unwrap()
+	if err != nil {
+		return err
+	}
+	inputs := inputsAny.(map[string]any)
+
+	blockhash, err := cap.reader.LatestBlockhash()
+
+	programID := config.ForwarderProgramID
+	state := config.ForwarderStateAccount
+	authority := config.FromAddress
+
+	// Determine store authority
+	// TODO: compute this only once per capability init
+	seeds := [][]byte{[]byte("forwarder"), state.Bytes()}
+	forwarderAuthority, _, err := sdk.FindProgramAddress(seeds, programID)
+	if err != nil {
+		return errors.Wrap(err, "error on FindProgramAddress")
+	}
+
+	data := []byte{}
+
+	// No signature validation in the MVP demo
+	signatures := [][]byte{} // TODO: validate each sig is 64 bytes
+
+	data = append(data, uint8(len(signatures))) // length prefix
+	for _, sig := range signatures {
+		data = append(data, sig...)
+	}
+
+	// TODO: encode inputs into data
+	data = append(data, inputs["report"].([]byte)...)
```

Comment:
looks like we are missing the evaluateParams step from the EVM side. Any reason this isn't included the same way it was done on EVM? Seems weird to pull the data from "report" vs the more complex eval we are doing on EVM. If that's what the TODO is for, no worries :)